test/UriTest.php

```diff
@@ -159,22 +158,23 @@ public function testWithPortReturnsNewInstanceWithProvidedPort($port)
     public function testWithPortReturnsSameInstanceWithProvidedPortIsSameAsBefore()
     {
         $uri = new Uri('https://user:pass@local.example.com:3001/foo?bar=baz#quz');
-        $new = $uri->withPort('3001');
+        $new = $uri->withPort(3001);
         $this->assertSame($uri, $new);
-        $this->assertEquals('3001', $new->getPort());
+        $this->assertEquals(3001, $new->getPort());
     }
 
     public function invalidPorts()
     {
         return [
-            'true'      => [ true ],
-            'false'     => [ false ],
-            'string'    => [ 'string' ],
-            'array'     => [ [ 3000 ] ],
-            'object'    => [ (object) [ 3000 ] ],
-            'zero'      => [ 0 ],
-            'too-small' => [ -1 ],
-            'too-big'   => [ 65536 ],
+            'true'       => [ true ],
+            'false'      => [ false ],
+            'string'     => [ 'string' ],
+            'string-int' => [ '3000' ],
+            'array'      => [ [ 3000 ] ],
+            'object'     => [ (object) [ 3000 ] ],
```

Comment:
This one is invalid, and needs to be changed regardless of other changes. It should read:

```php
'object' => [ (object) ['port' => 3000]],
```